system-tests/e2e-suite/src/test/scala/com/hotels/styx/admin/FileBasedOriginsFileChangeMonitorSpec.scala

```diff
@@ -0,0 +1,123 @@
+
+import java.io.ByteArrayInputStream
+import java.nio.charset.StandardCharsets.UTF_8
+import java.nio.file.Files.{copy, delete}
+import java.nio.file.StandardCopyOption.REPLACE_EXISTING
+import java.nio.file.{Files, Path, Paths}
+
+import com.google.common.io.Files.createTempDir
+import com.hotels.styx.api.messages.FullHttpRequest.get
+import com.hotels.styx.api.messages.HttpResponseStatus
+import com.hotels.styx.support.api.BlockingObservables.waitForResponse
+import com.hotels.styx.support.backends.FakeHttpServer
+import com.hotels.styx.support.configuration._
+import com.hotels.styx.{StyxClientSupplier, StyxServerSupport}
+import org.scalatest.concurrent.Eventually
+import org.scalatest.{BeforeAndAfterAll, ConfigMap, FunSpec, ShouldMatchers}
+
+class FileBasedOriginsFileChangeMonitorSpec extends FunSpec
+  with StyxServerSupport
+  with StyxClientSupplier
+  with ShouldMatchers
+  with BeforeAndAfterAll
+  with Eventually {
+
+  val tempDir = createTempDir()
+  val styxOriginsFile = Paths.get(tempDir.toString, "origins.yml")
+
+  val origin = FakeHttpServer.HttpStartupConfig(appId = "app", originId="app").start()
+
+  // TODO: Mikko: Investigate why changing the path fails!
+
+  val configTemplate =
+    """---
+      |- id: "%s"
+      |  path: "%s"
+      |  connectionPool:
+      |    maxConnectionsPerHost: 45
+      |    maxPendingConnectionsPerHost: 15
+      |    socketTimeoutMillis: 120000
+      |    connectTimeoutMillis: 1000
+      |    pendingConnectionTimeoutMillis: 8000
+      |  healthCheck:
+      |    uri: "/version.txt"
+      |    intervalMillis: 5000
+      |  responseTimeoutMillis: 60000
+      |  origins:
+      |  - { id: "app1", host: "localhost:%d" }
+      |""".stripMargin
+
+  private val yamlConfig: String =
+    s"""---
+       |proxy:
+       |  connectors:
+       |    http:
+       |      port: 0
+       |
+       |admin:
+       |  connectors:
+       |    http:
+       |      port: 0
+       |
+       |services:
+       |  factories:
+       |    backendServiceRegistry:
+       |      class: "com.hotels.styx.proxy.backends.file.FileBackedBackendServicesRegistry$$Factory"
+       |      config:
+       |        originsFile: "${styxOriginsFile.toString}"
+       |        monitor:
+       |          enabled: true
+    """.stripMargin
+
+  writeConfig(styxOriginsFile, configTemplate.format("appv1", "/app01/", origin.port()))
+
+  val styxServer = StyxYamlConfig(yamlConfig = yamlConfig).startServer()
+  val reqToApp01 = get(s"http://localhost:${styxServer.proxyHttpAddress().getPort}/app01/x").build()
+  val reqToApp02 = get(s"http://localhost:${styxServer.proxyHttpAddress().getPort}/app02/x").build()
+
+
+  override protected def beforeAll(configMap: ConfigMap) {
+    styxServer.isRunning should be(true)
+  }
+
+  it("Responds with BAD_REQUEST when the origins cannot be read") {
+    waitForResponse(client.sendRequest(reqToApp01.toStreamingRequest)).status() should be (HttpResponseStatus.OK)
```

Comment:
We should import these status codes so we don't need to include the class name.

Reply:
Changed the test case name.